### PR TITLE
Build: Upgrade grunt-contrib-qunit to 3.x (Headless Chrome)

### PR DESCRIPTION
**Pull request by Krinkle**
Ref #1302.

### CODE REVIEW

**Comment by Krinkle** [comment]
The tests pass, except for `test/amd.html`. I'm not sure what's going on there.

**Comment by Arkni** [comment]
@Krinkle 

The local bridge file used for collecting coverage should also be changed to use `[self|window].__grunt_contrib_qunit__` instead of `window.alert`:
https://github.com/qunitjs/qunit/blob/9af04486aaf1bf23b602a8f379a7826989a9c89e/build/coverage-bridge.js#L5-L9

See: https://github.com/gruntjs/grunt-contrib-qunit/blob/a4be5b5b41bc8df550cf3668be9b973a200deae9/chrome/bridge.js#L23-L25

**Comment by Krinkle** [comment]
@Arkni Thank for catching that!

**Comment by platinumazure** [comment]
The code here looks good, but I need to see what was done in grunt-contrib-qunit as well before I can sign off. Hopefully I'll have time to do that this week.

File: build/coverage-bridge.js [hunk 1]
```diff
@@ -4,8 +4,7 @@
 
 	// Send messages to the parent PhantomJS process via alert! Good times!!
```
Comment (platinumazure) [inline build/coverage-bridge.js]: This comment seems inaccurate. Should it be removed?

**Comment by Arkni** [comment]
Hi @Krinkle,

I think the amd.html test fails because the coverage bridge was not designed to work with amd just like the built-in bridge.

Also, after spending some time navigating the new code of `grunt-contrib-qunit` in the hope of making it more functional as it was with `PhantomJS` or even better, I found that it reports wrong stats for QUnit tests:
* Before upgrading:
```
287 tests completed with 0 failed, 15 skipped, and 5 todo. 
830 assertions (in 2414ms), passed: 825, failed: 5
```
* After:
```
324 tests completed with 2 failed, 18 skipped, and 9 todo. 
922 assertions (in 2414ms), passed: 907, failed: 15
```

The two failed tests are:
1. The test `Unhandled Rejections inside test context - test passes just fine, but has a rejected promise` in `/test/reporter-html/unhandled-rejection.js`. And
2. The test `error is thrown when overloading QUnit global` in `/test/overload.html` as Chrome reports the generic error `Script error.` instead of the raised error `QUnit has already been defined.` due to this rule https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror#Notes but this only happens when using the file URI scheme (works fine in Firefox and PhantomJS).

For now, I wouldn't recommend landing this PR untill I or someone else found the time to fix `grunt-contrib-qunit` (it was my fault I didn't review the migration PR, but trying my best to fix it).

**Comment by trentmwillis** [comment]
Thanks all for working on this! Just wanted to call out that this would also fix https://github.com/qunitjs/qunit/issues/1172.

**Comment by trentmwillis** [comment]
I've rebased this branch and fixed the test failures; the AMD test failures were kind of tricky but basically we tried to set up the bridge before the QUnit global was available. Stole a solution from the `grunt-contrib-qunit` lib: https://github.com/gruntjs/grunt-contrib-qunit/blob/a4be5b5b41bc8df550cf3668be9b973a200deae9/chrome/bridge.js#L10-L16

**Pull request merged**